### PR TITLE
feat: restores all qCalcCondition for chart conversion

### PR DESCRIPTION
**Pull request by LiKang6688**
chart conversion task 1:

- enable conditional show for chart conversion by `conversion.conditionalShow.unquarantine(propertyTree.qProperty);`	
	-	when a column is hidden in an old table, after converting that old table to the new one, that column remains hidden and vice versa.

chart conversion task 2: column width will be in the next PR

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -11,24 +11,26 @@ export interface TextAlign {
...
 }
 interface InlineMeasureDef extends EngineAPI.INxInlineMeasureDef {
   textAlign: TextAlign;
+  columnWidth?: ColumnWidth;
 }
 interface AttributeExpressionProperties extends EngineAPI.INxAttrExprDef {
   id: 'cellForegroundColor' | 'cellBackgroundColor';
 }
-interface DimensionProperties extends Omit<EngineAPI.INxDimension, 'qDef' | 'qAttributeExpressions'> {
+export interface DimensionProperty extends Omit<EngineAPI.INxDimension, 'qDef' | 'qAttributeExpressions'> {
```
Comment (haxxmaxx) [inline src/types.ts:23]: why renaming this? and does it need to be exported?

File: src/types.ts [hunk 2]
```diff
@@ -11,24 +11,26 @@ export interface TextAlign {
 // properties
 interface InlineDimensionDef extends EngineAPI.INxInlineDimensionDef {
   textAlign: TextAlign;
+  columnWidth?: ColumnWidth;
```
Comment (haxxmaxx) [inline src/types.ts:14]: Are these really optional?

File: src/types.ts (hunk 1 above)
Comment (LiKang6688) [reply src/types.ts:23]: they are needed to be exported for the next PR, remove for now

File: src/types.ts (hunk 2 above)
Comment (LiKang6688) [reply src/types.ts:14]: yes, for the new table, it is optional, remove for now

File: package.json [hunk 3]
```diff
@@ -138,7 +138,8 @@
     "stylis-plugin-rtl-sc": "npm:stylis-plugin-rtl@1.1.0",
     "ts-jest": "29.1.0",
     "typescript": "5.0.3",
-    "yargs": "17.7.1"
+    "yargs": "17.7.1",
+    "qlik-object-conversion": "0.16.0"
```
[the remark this answers is not in the document]
Comment (LiKang6688) [reply package.json]: updated

File: .eslintrc.json [hunk 4]
```diff
@@ -27,11 +27,11 @@
     "react/jsx-props-no-spreading": 0,
     "react-hooks/rules-of-hooks": 2,
     "react-hooks/exhaustive-deps": 1,
-    "import/no-extraneous-dependencies": [2, { "devDependencies": true }],
-    "prefer-destructuring": ["error", { "object": true, "array": false }],
+    "import/no-extraneous-dependencies": 0,
```
Comment (cbt1) [inline .eslintrc.json:30]: Did you need to disable this rule?

Comment (LiKang6688) [reply .eslintrc.json]: Good find, it can be enabled now

**Review by cbt1 — APPROVED** [review]
Have not tested it myself but LGTM. I do think it allows us to fix other conversion related issues later, like column width for exmaple.

File: src/ext/conversion/importProperties.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import conversion from 'qlik-object-conversion';
+
+import data from '../../qae/data';
+import { ExportFormat, PropTree } from '../../types';
+
+const importProperties = (
+  exportFormat: ExportFormat,
+  initialProperties: EngineAPI.IGenericHyperCubeProperties,
+  extension: any,
```
Comment (cbt1) [inline src/ext/conversion/importProperties.ts:9]: Was it not possible to use `unknown` instead?

Comment (LiKang6688) [reply src/ext/conversion/importProperties.ts:9]: as there is 
```
defaultPropertyValues: {
      defaultDimension: extension.getDefaultDimensionProperties(),
      defaultMeasure: extension.getDefaultMeasureProperties(),
    },
```

**Pull request merged**